### PR TITLE
First attempt to minimize to system tray

### PR DESCRIPTION
**Pull request by joshnovak**
I too am completely annoyed that the hangouts desktop app is disappearing.
Here's a (hopefully) small change to minimize the app to the system tray.

### CODE REVIEW

**Review by brmonaghan — COMMENTED** [review]
Will look at this.

**Comment by brmonaghan** [comment]
Still Not working. Not Approving till this works

**Comment by joshnovak** [comment]
Works for me - I wanted to share my changes.  Maybe it's your environment.  
If you let me know what you are seeing, I can try to fix.

**Comment by brmonaghan** [comment]
I launch voice with npm in git, it launches and when I close it, it closes with no notification.

Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Comment by Jerrkawz** [comment]
@brmonaghan  Works for me. Its not close its minimize that sends it to the tray but i think it should probably be close.

@joshnovak 
Few issues though:
1. I have two computers I'm testing on. It works fine on windows 10, but it doesn't work on mac because mac wont scale the image. You need to take the 1024px image and scale it to 16x16 and use that one for the tray. 

2. It sends to tray on minimize, i think id rather it send to try on close.

3. Clicking on the notification doesnt unhide the window.

Number 3 might be a bit more difficult to fix, so if you cant figure that one out then just fix 1 and 2 and we can create an issue for 3 to fix later

File: src/main.js [hunk 1]
```diff
@@ -1,12 +1,14 @@
-const { app, nativeImage, BrowserWindow } = require('electron')
+const { app, nativeImage, BrowserWindow, Tray, Menu } = require('electron')
 const BadgeGenerator = require('./badge_generator')
 const REFRESH_RATE = 5000; // 5 seconds
+const iconPNG = "1024px-Google_Voice_icon_(2020).png";
 const icon = nativeImage.createFromPath(
-    app.getAppPath() + "/public/1024px-Google_Voice_icon_(2020).png"
```
Comment (Jerrkawz) [inline src/main.js:5]: Need to create a new 16x16 image and use that one for the tray, but keep the big one for the dock / task bar

File: src/main.js [hunk 2]
```diff
@@ -27,6 +29,18 @@ function createWindow() {
 
     currentInterval = setInterval(updateNotifications.bind(this, app, win), REFRESH_RATE);
 
+    // Add support for tray icon
+    win.on('minimize', function (event) {
```
Comment (Jerrkawz) [inline src/main.js:33]: change to on('close') please

Comment (Jerrkawz) [reply src/main.js:33]: You will have to delete line 51-56 though if we use close, you should remove the app.on('window-all-closed') handler

**Comment by Jerrkawz** [comment]
@joshnovak Oh thought of one more thing. Theres no way in the notification pain to see if there are unread messages so i propose the following:

Put a hook into my setInterval that also updates the notification image when the count > 0. Use another 16x16 voice logo except with a small red dot in the bottom right corner, maybe roughly a 6x6 or an 8x8. whatever looks best

**Comment by joshnovak** [comment]
I was going to argue about #2 - but after thinking about it, this is how the hangouts currently works, so I accepted it.
I changed #1 and #2 - hopefully they are fixed.

I'm not worried about the rest, so I think separate issues would be better to track those.

**Comment by brmonaghan** [comment]
Approve.

**Review by brmonaghan — APPROVED** [review]
Now works when closing.

**Review by brmonaghan — APPROVED** [review]
Now works when closing.

**Comment by Jerrkawz** [comment]
Thanks! Merged

**Pull request merged**